railties/lib/rails/command/base.rb

```diff
@@ -0,0 +1,135 @@
+require "thor"
+require "erb"
+
+require "active_support/core_ext/string/filters"
+require "active_support/core_ext/string/inflections"
+
+require "rails/command/actions"
+
+module Rails
+  module Command
+    class Base < Thor
+      class Error < Thor::Error # :nodoc:
+      end
+
+      include Actions
+
+      class << self
+        # Returns true when the app is a Rails engine.
+        def engine?
+          defined?(ENGINE_ROOT)
+        end
+
+        # Tries to get the description from a USAGE file one folder above the command
+        # root.
+        def desc(usage = nil, description = nil)
```

Comment:
The `desc` is from Thor: https://github.com/erikhuda/thor/blob/06dade7c9567edccc4da2f0d6fcf4c2d00066e3d/lib/thor.rb#L54

We could add an alias for this description, which I agree would be more... descriptive 🤓